spec/lib/flapjack/data/contact_spec.rb

```diff
@@ -275,20 +277,26 @@
     expect(credentials).not_to be_nil
     expect(credentials).to be_a(Hash)
     expect(credentials).to eq({'service_key' => '123456789012345678901234',
-                           'subdomain'   => 'flpjck',
-                           'token'       => 'token123'})
+                               'subdomain'   => 'flpjck',
+                               'token'       => 'token123',
+                               'username'    => '',
+                               'password'    => ''})
   end
 
   it "sets pagerduty credentials for a contact" do
     contact = Flapjack::Data::Contact.find_by_id('c362', :redis => @redis)
     contact.set_pagerduty_credentials('service_key' => '567890123456789012345678',
                                       'subdomain'   => 'eggs',
-                                      'token'       => 'token123')
+                                      'token'       => 'token123',
```

Comment:
Yes. The idea being you can set either token, or username and password, or both if you want to. The only logic in flapjack to select between one and the other is in the pagerduty gateway where it'll use token, or fall back to username and password. It seemed simplest to implement it this way. It means there should be no friction when either staying with your existing username+password auth, or switching to token (just add the token).